### DISCUSSION TITLE
Overall map view

### DISCUSSION THREAD

**Discussion by MrNickIE** [teslamate-org/teslamate#3435] [Ideas]
We use Teslamate to manage a fleet of 10 vehicles.  

We would like to have a single map view we can put on a screen in the office that shows the location of all the teslas that we oversee for the team.  Preferably a URL that allows us to display full screen without borders, with a set zoom level and centralised location (so we can load it up with a signage product)

**Comment by DrMichael** [comment]
Interesting
I have created a dashboard showing the last position of each car as repeating panels.
I did not figure out yet, how to get the last positions for all cars with one query though. Perhaps @Dulanic has an idea?

**Reply by Dulanic** [reply]
Maybe, but I'm traveling on a 20 hour flight today so it won't be soon lol

**Comment by DrMichael** [comment]
Well, I would need a database with several cars to procede... I got an idea, but cannot really test it.

**Reply by MrNickIE** [reply]
I could give some kind of temp access to our system if needed

**Reply by DrMichael** [reply]
That would help. How could we do that?

**Reply by MrNickIE** [reply]
Thats a great question, for which I do not have an answer.  I was trying to figure out how to privately message on this thing....

**Comment by DrMichael** [comment]
Ok, made a fake car in my data... This appears to work, but is slow like hell... Pls try it.
[Car Locations-1700152654145.json](https://github.com/teslamate-org/teslamate/files/13380248/Car.Locations-1700152654145.json)

**Reply by MrNickIE** [reply]
Not sure how to use this :)  (Noob user, remember) :)

**Reply by DrMichael** [reply]
Goto Dashboards, then New-> Import and Upload the file...

**Reply by MrNickIE** [reply]
<img width="231" alt="image" src="https://github.com/teslamate-org/teslamate/assets/82377682/04b41c4f-a3fb-4960-93e1-5671fad643cb">

I believe it is looking for a datasource not on my system

**Comment by DrMichael** [comment]
Ok, what does it say, when you move the mouse over the red field?

**Reply by MrNickIE** [reply]
<img width="314" alt="image" src="https://github.com/teslamate-org/teslamate/assets/82377682/8a2862ee-3d28-4b68-9b71-e5323dd9ca65">

**Reply by DrMichael** [reply]
Ok, click Edit in the dashboard
![image](https://github.com/teslamate-org/teslamate/assets/8087682/cbe6188d-fb73-4f3f-88d4-213acca8f905)

Then you can set the datasource in
![image](https://github.com/teslamate-org/teslamate/assets/8087682/54139b2a-61b4-4803-998f-4d990cfdc24a)

**Comment by MrNickIE** [comment]
How cool!

![image](https://github.com/teslamate-org/teslamate/assets/82377682/50143654-be3d-47f7-aa90-a39e7a55820e)

It doesnt maintain the zoom yet, but this is class.  Font sizing needs work, but this is a great start!  Thank you so much!

**Comment by DrMichael** [comment]
> It doesnt maintain the zoom yet, ...

What do you mean?

> Font sizing needs work

What do you propose?

> Thank you so much!

It was a fun challenge... :-)

**Comment by MrNickIE** [comment]
Hi @DrMichael 

Again, thanks for this.

The Font text size of the vehicle name, would like to be able to adjust this, but I am sure I will figure it out myself.
Regarding the zoom, I also believe I can do this within the template.

**Comment by DrMichael** [comment]
If you're done, could you share it then. I got the feeling that there are more people interested.
We could include it in the Grafana dashboards.

**Reply by MrNickIE** [reply]
Of course.   Just playing around with settings a bit first.

**Reply by MrNickIE** [reply]
So, the biggest issue I have right now is that when the page refreshes, its cutting off the bottom of the map, even with a larger padding set. 

On my larger monitor (huge one) it displays, but on a regular 20' screen it does not shrink the map to fit within the window.

**Reply by MrNickIE** [reply]
Interestingly, when editing the dashboard, there is a "fill and actual" toggle at the top, and this is what is needed in the live view to fix this (but it does not seem to exist)

**Reply by MrNickIE** [reply]
[Car Locations (All)-1700311069901.json](https://github.com/teslamate-org/teslamate/files/13400271/Car.Locations.All.-1700311069901.json)
My edits

**Reply by MrNickIE** [reply]
Last update:  I would recommend this dashboard is set to last 3 hours and 1 min refresh.  Seems to be smoother experience on my system
![image](https://github.com/teslamate-org/teslamate/assets/82377682/951b3837-041f-472e-a9ce-a439b08a7294)

**Reply by DrMichael** [reply]
right, that should obviously be faster. I had set it to 3 days, because my car wakes up all appr. 2,5 days...

**Reply by DrMichael** [reply]
Isn't fontsize 10 a bit small?

**Reply by MrNickIE** [reply]
Works for me and I'm pretty old now 😂. I found that if it's more and they are in a cluster over the city the names overlap.

**Comment by DrMichael** [comment]
Are there more people interested in a fleet overview?

**Comment by MrNickIE** [comment]
Hi again,

So, the dashboard view has been working well, but now I am being asked if there is some way to have this page:
![image](https://github.com/teslamate-org/teslamate/assets/82377682/931a4929-5b0e-49f9-b8a8-e41d3dc59e64)

But instead of separate windows, we have a whole page that dynamically updates?

I have no idea if this is even possible, but hey....   I am the messenger :)
